src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AssemblyCatalog.cs

```diff
@@ -0,0 +1,603 @@
+        ///     Initializes a new instance of the <see cref="AssemblyCatalog"/> class 
+        ///     with the specified code base.
+        /// </summary>
+        /// <param name="codeBase">
+        ///     A <see cref="String"/> containing the code base of the assembly containing the
+        ///     attributed <see cref="Type"/> objects to add to the <see cref="AssemblyCatalog"/>.
+        /// </param>
+        /// <param name="reflectionContext">
+        ///     The <see cref="ReflectionContext"/> a context used by the catalog when 
+        ///     interpreting the types to inject attributes into the type definition<see cref="AssemblyCatalog"/>.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="codeBase"/> is <see langword="null"/>.
+        ///     <para>
+        ///         -or-
+        ///     </para>
+        ///     <paramref name="reflectionContext"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///     <paramref name="codeBase"/> is a zero-length string, contains only white space, 
+        ///     or contains one or more invalid characters. />.
+        /// </exception>
+        /// <exception cref="PathTooLongException">
+        ///     The specified path, file name, or both exceed the system-defined maximum length. 
+        /// </exception>
+        /// <exception cref="SecurityException">
+        ///     The caller does not have path discovery permission. 
+        /// </exception>
+        /// <exception cref="FileNotFoundException">
+        ///     <paramref name="codeBase"/> is not found.
+        /// </exception>
+        /// <exception cref="FileLoadException ">
+        ///     <paramref name="codeBase"/> could not be loaded.
+        ///     <para>
+        ///         -or-
+        ///     </para>
+        ///     <paramref name="codeBase"/> specified a directory.
+        /// </exception>
+        /// <exception cref="BadImageFormatException">
+        ///     <paramref name="codeBase"/> is not a valid assembly
+        ///     -or- 
+        ///     Version 2.0 or later of the common language runtime is currently loaded 
+        ///     and <paramref name="codeBase"/> was compiled with a later version. 
+        /// </exception>
+        /// <remarks>
+        ///     The assembly referenced by <paramref langword="codeBase"/> is loaded into the Load context.
+        /// </remarks>
+        public AssemblyCatalog(string codeBase, ReflectionContext reflectionContext)
+        {
+            Requires.NotNullOrEmpty(codeBase, "codeBase");
+            Requires.NotNull(reflectionContext, "reflectionContext");
+
+            InitializeAssemblyCatalog(LoadAssembly(codeBase));
+            this._reflectionContext = reflectionContext;
+            this._definitionOrigin = this;
+        }
+#endif //FEATURE_REFLECTIONCONTEXT && FEATURE_REFLECTIONFILEIO
+
+
+//#if FEATURE_REFLECTIONFILEIO
```

Comment:
All these commented out defines can be removed (retaining the content) ie search for `^//#.*\n`